### DISCUSSION TITLE
DCAT and ODRL based information model for IDS (and beyond)

### DISCUSSION THREAD

**Discussion by ssteinbuss** [International-Data-Spaces-Association/IDS-G#41] [Ideas]
We discussed about the use of ODRL and DCAT as a foundation for the IDS Infomodel. Let's get the content together here and contribute to the issue

**Comment by clange** [comment]
Not sure what exactly you would like to see. From the very beginning, the IDS infomodel has been based on DCAT and ODRL. With the recent release 4.2 (already in main; will be announced once the Java bindings have been generated), we have made the reuse even more explicit, in an effort led by @JohannesLipp.

**Reply by sebbader** [reply]
I think the main issue has been the usage of the IDS namespace instead of the pure ODRL and DCAT one, e.g.:
```
ids:ContractAgreement  rdfs:subClassOf  odrl:Agreement .
```
instead of directly using `odrl:Agreement`. It feels like it would be great to use this discussion to explain why the pattern with the IDS namespace has been used to reach a wider acceptance.

**Reply by JohannesLipp** [reply]
Please note that we recently tackled exactly this issue, where we replaced IDS-duplicates with direct use of such external concepts. For a full documentation please see [this readme file](https://github.com/International-Data-Spaces-Association/InformationModel/blob/e17a4cfce1cdfcf2e2e72595b5c2c03c6cf96b75/evaluation_external/README.md).

[Example diff](https://github.com/International-Data-Spaces-Association/InformationModel/pull/504/files?authenticity_token=kGiydeAQ8UlRAisARdTUJhjnw0k6V0Wk9KyB9dcsDDlTY2vVfJLrCtSS4qFerjNL%2BtMAPideMZg7ck5wdhyQOQ%3D%3D&file-filters%5B%5D=.html&file-filters%5B%5D=.json&file-filters%5B%5D=.jsonld&file-filters%5B%5D=.md&file-filters%5B%5D=.n3&file-filters%5B%5D=.nt&file-filters%5B%5D=.rdf&file-filters%5B%5D=.ttl&file-filters%5B%5D=.xml&hide-deleted-files=true#diff-9bfad5a86fbcef77ddbcdeba01b31e0df26316877265ddf7af69d2b50e2d5489):
![image](https://user-images.githubusercontent.com/15930331/189625499-9c9e7cd5-bb5d-4a7e-aef0-fef60f6c707a.png)

Full results: All files with "shape" in their name [from this line-up](https://github.com/International-Data-Spaces-Association/InformationModel/pull/504/files?authenticity_token=z2n2iOs%2BObXzkzt7YgeZ%2BI8SUstTrUSP37i5f4hWqvsMYi8od7wj9nYD8tp5fX6VbSaRvE6kMLMQZnb6KWY2%2Bw%3D%3D&file-filters%5B%5D=.ttl&w=1).

More concrete example:
![image](https://user-images.githubusercontent.com/15930331/189625974-215f0201-a8a0-43b1-816d-9688b0850d66.png)

**Reply by JohannesLipp** [reply]
The changes are partially published through new versions of the information model. The rest is still pending, as we need to incorporate them in a new version due to breaking changes (i.e., change ids:MediaType to dct:MediaType).

**Comment by sebbader** [comment]
At the core, the IDS interprets the ODRL and DCAT concepts in a bit stricter manner than the original creators of the vocabularies wanted. This is simply due to the fact that IDS Connectors have pretty clear requirements while the general vocabularies shall be applicable in many use cases. Therefore, the original definitions have been extended with more details in the IDS IM. 

Now, if we apply these details to e.g. `odrl:Agreement` directly, we would overwrite/extend the standard itself. That's not a good practice, as we do not control these vocabularies. The convention is to define subclasses in our own namespace, and put the explanations to them. In general, as @clange has mentioned, this means that
a) the IDS IM *is* based on ODRL and DCAT, and
b) the best practice to do this has been applied.

**Comment by sebbader** [comment]
@PeterKoen-MSFT would be great if you can add your points to this discussion.
